client_test/rpc_test.go

```diff
@@ -44,143 +46,244 @@ func createLogger(logDestination *os.File, clientName, rpcRole string) zerolog.L
-		expectedPaymentInfo(vRes.ChannelId, simpleOutcome(ta.Alice.Address(), ta.Bob.Address(), 100, 0), query.Open),
-		expectedPaymentInfo(vRes.ChannelId, simpleOutcome(ta.Alice.Address(), ta.Bob.Address(), 99, 1), query.Open),
-		expectedPaymentInfo(vRes.ChannelId, simpleOutcome(ta.Alice.Address(), ta.Bob.Address(), 99, 1), query.Complete),
+	laiClosure := aliceClient.CloseLedger(aliceLedger.ChannelId)
+	<-aliceClient.ObjectiveCompleteChan(laiClosure)
+	// bobClient.CloseLedger(bobLedger.ChannelId)
+
+	if n != 2 { // for n=2, alice and bob share a ledger, which should only be closed once.
+		libClosure := bobClient.CloseLedger(bobLedger.ChannelId)
+		<-bobClient.ObjectiveCompleteChan(libClosure)
 	}
-	optionalVirtualNotifs := []query.PaymentChannelInfo{
-		expectedPaymentInfo(vRes.ChannelId, simpleOutcome(ta.Alice.Address(), ta.Bob.Address(), 99, 1), query.Closing),
-		// TODO: Sometimes we see a closing notification with the original balance.
-		// See https://github.com/statechannels/go-nitro/issues/1306
-		expectedPaymentInfo(vRes.ChannelId, simpleOutcome(ta.Alice.Address(), ta.Bob.Address(), 100, 0), query.Closing),
+
+	//////////////////////////
+	// perform wrap-up checks
+	//////////////////////////
+
+	for i, client := range clients {
+		if i != 0 {
+			leftLC := ledgerChannels[i-1]
+			vcCount := len(client.GetPaymentChannelsByLedger(leftLC.ChannelId))
+			if vcCount != 0 {
+				t.Errorf("expected no virtual channels in ledger channel %s, got %d", leftLC.ChannelId, vcCount)
+			}
+		}
+		if i != n-1 {
+			rightLC := ledgerChannels[i]
+			vcCount := len(client.GetPaymentChannelsByLedger(rightLC.ChannelId))
+			if vcCount != 0 {
+				t.Errorf("expected no virtual channels in ledger channel %s, got %d", rightLC.ChannelId, vcCount)
+			}
+		}
 	}
-	checkNotifications(t, requiredVirtualNotifs, optionalVirtualNotifs, aliceVirtualNotifs, defaultTimeout)
 
-	checkNotifications(t, requiredVirtualNotifs, optionalVirtualNotifs, bobVirtualNotifs, defaultTimeout)
+	aliceLedgerNotifs := aliceClient.LedgerChannelUpdatesChan(ledgerChannels[0].ChannelId)
+	expectedAliceLedgerNotifs := createLedgerStory(
+		aliceLedger.ChannelId, alice.Address(), actors[1].Address(), // actor[1] is the first intermediary - can be Bob if n=2 (0-hop)
+		[]channelStatusShorthand{
+			{100, 100, query.Proposed},
+			{100, 100, query.Open},
+			{0, 100, query.Open},
+			{99, 101, query.Open},
+			{99, 101, query.Closing},
+			{99, 101, query.Complete},
+		},
+	)
+	checkNotifications(t, "aliceLedger", expectedAliceLedgerNotifs, []query.LedgerChannelInfo{}, aliceLedgerNotifs, defaultTimeout)
+
+	bobLedgerNotifs := bobClient.LedgerChannelUpdatesChan(bobLedger.ChannelId)
+	expectedBobLedgerNotifs := createLedgerStory(
+		bobLedger.ChannelId, actors[n-2].Address(), bob.Address(),
+		[]channelStatusShorthand{
+			{100, 100, query.Proposed},
+			{100, 100, query.Open},
+			{0, 100, query.Open},
+			{99, 101, query.Open},
+			{99, 101, query.Complete},
```

Comment:
It's really helpful to see all of the expected notifications listed out succinctly.